File: CS50/phonebook/phonebook.c
#include <stdio.h>
#include <string.h>

typedef struct
{
    char name[50];
    char number[20];
} person;

int main(void)
{
    person people[3];

    strcpy(people[0].name, "David");
    strcpy(people[0].number, "+1-617-495-100");

    strcpy(people[1].name, "John");
    strcpy(people[1].number, "+1-949-468-2750");

    strcpy(people[2].name, "Yulia");
    strcpy(people[2].number, "+1-617-495-100");

    char name[50];
    printf("Name: ");
    fgets(name, sizeof(name), stdin);

    // Remove newline character if present
    name[strcspn(name, "\n")] = '\0';

    for (int i = 0; i < 3; i++)
    {
        if (strcmp(people[i].name, name) == 0)
        {
            printf("Found %s\n", people[i].number);
            return 0;
        }
    }

    printf("Not found\n");
    return 1;
}

/// This program demonstrates how to use a struct to store information about people, including their names and phone numbers.
/// It uses an array of structs to store multiple people's information and allows the user to search for a person's phone number by name.
/// The program uses the fgets function to read a line of input from the user and stores it in a character array.
/// It also uses the strcmp function to compare strings and the strcspn function to remove the newline character from the input.
/// The program is a simple example of how to use structs and arrays in C to store and manipulate data.
/// It also shows how to use the fgets function to read strings safely and how to manipulate strings using functions like strcspn and strcmp.
/// The program is a simple demonstration of how to use structs and arrays in C to store and manipulate data.

File: CS50/phonebook_py_v2/phonebook.py
people = {
    "Maria": "+1-617-495-1000",
    "Sofia": "+1-617-495-1001",
    "Lauren": "+1-617-495-1002",
}

name = input("name: ")

if name in people:
    print(f"Number: {people[name]}")
else:
    print("Not found")

# This code checks if a given name is in a predefined dictionary of names and phone numbers.
# It prompts the user for a name and checks if that name is in the dictionary.
# If the name is found, it prints the corresponding phone number; otherwise, it prints "Not found."
# The code is a simple demonstration of dictionary membership testing in Python.